test/integration/application_controller_test.rb

```diff
@@ -44,6 +44,17 @@ def setup
     get admin_buyers_accounts_path
   end
 
+  test "proxy config objects tracked for changes are cleared" do
+    provider = FactoryBot.create(:provider_account)
+    login! provider
+
+    ProxyConfigAffectingChanges::Tracker.any_instance.expects(:reported_clear)
+    # make sure #reported_clear is not called by #flush_proxy_affecting_changes
+    ApplicationController.any_instance.expects(:flush_proxy_affecting_changes)
```

Comment:
It is expected to be called once. Why do you think it is any number of times?

The test would work even if `flush_proxy_affecting_changes` is not stubbed. But I stub it, to make sure it's code is not executed. So that makes sure the call to `reported_clear` is not called by it.

Not sure where the misunderstanding comes from and whether I actually answered the question.

Reply:
> It is expected to be called once. Why do you think it is any number of times?

Yes you're right. `expects` means expected once, `stubs` means expected any number of times.